docs/050-breaking-changes.rst

```diff
@@ -160,38 +169,52 @@ New version:
       OtherContract other;
       uint myNumber;
 
-      // Function visibility not provided, not an error
-      // Function mutability not provided, not an error
+      // Function visibility must be specified.
+      // Function mutability must be specified.
       function f(uint x) public returns (bytes memory) {
-         int y = -3 >> 1; // y = -2 (correct)
-
+         int y = -3 >> 1;
+         // y = -2 (correct)
          do {
             x += 1;
-            if (x > 10) continue; // jumps to the condition below
+            if (x > 10) continue;
+            // 'Continue' jumps to the condition below.
          } while (x < 11);
 
-         // Call returns (bool, bytes)
-         // Data location must be specified
+         // Call returns (bool, bytes).
+         // Data location must be specified.
          (bool success, bytes memory data) = address(other).call("f");
          if (!success)
             revert();
          return data;
       }
 
       using address_make_payable for address;
-      function g(uint[] memory arr, bytes8 x, OtherContract otherContract) public payable { // No need to explicit parameter location data
-         // 'contract.transfer' is not provided
-         // 'address(contract).transfer' is not provided since 'address(contract)' is not 'address payable'.
-         // So we need to convert variable 'contract' to type 'address payable'.
-         address payable addr = address(otherContract).make_payable();
+      // Data location for 'arr' must be specified
+      function g(uint[] memory arr, bytes8 x, OtherContract otherContract, address unknownContract) public payable {
+         // 'otherContract.transfer' is not provided.
+         // Since the code of 'OtherContract' is known and has the fallback
+         // function, address(otherContract) has type 'address payable'.
+         address(otherContract).transfer(1 ether);
+
+         // 'unknownContract.transfer' is not provided.
+         // 'address(unknownContract).transfer' is not provided
+         // since 'address(unknownContract)' is not 'address payable'.
+         // If the function takes an 'address' that you know it is the
```

Comment:
``takes an 'address' for which you know that it is the address of ...``?
But yes, in general I like it much better with these changes. Although it still doesn't really make sense here - if one knows that the arguments of a function will always be payable addresses, one would just change the argument type to ``address payable`` :-). But I think we can live with that - any case where we need the ``make_payable`` conversion will be highly artificial I guess.

Reply:
Actually no! There's an idea: instead of ``transfer`` we use ``unknownContract.make_payable().send(..)`` and check the return value here - that actually makes sense to do :-).

Reply:
Yea this example is kinda tricky... trying with `send` only

Reply:
Yes, true - ``send`` might somewhat make sense, though - the rationale being something like: our function works for payable and non-payable ``unknownContract``, hence it accepts type ``address``, not ``address payable``, but in case the contract is in fact payable, we want to send ether to it (for whatever reason), so we cast to ``address payable`` and use ``send()`` and check the return value, which will tell you whether the send succeeded. Probably no one will ever do something like that, but it's valid and makes sense in theory :-).